File: hospital-appointment-backend/src/main/java/com/hospital/appointment/entity/Doctor.java
package com.hospital.appointment.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 医生实体类
 * 新增
 */
@Data
@TableName("doctor")
public class Doctor implements Serializable {
    
    @TableId(type = IdType.AUTO)
    private Long doctorID;
    
    /** 医生姓名 */
    private String doctorName;
    
    /** 登录密码 */
    private String doctorPassword;
    
    /** 性别：1-男 2-女 */
    private Integer doctorGender;
    
    /** 手机号 */
    private String doctorPhone;
    
    /** 邮箱 */
    private String doctorEmail;
    
    /** 所属科室ID */
    private Long departmentID;
    
    /** 职称：主任医师 副主任医师 主治医师 住院医师 */
    private String title;
    
    /** 专长 */
    private String specialty;
    
    /** 医生简介 */
    private String doctorIntro;
    
    /** 医生照片URL */
    private String doctorPhoto;
    
    /** 挂号费 */
    private BigDecimal registrationFee;
    
    /** 医生状态：0-离职 1-在职 2-停诊 */
    private Integer doctorStatus;
    
    /** 创建时间 */
    private LocalDateTime createTime;
    
    /** 非数据库字段：科室名称 */
    @TableField(exist = false)
    private String departmentName;
}

File: hospital-appointment-frontend/src/api/hospital/doctor.ts
/**
 * 医生管理 API
 */

import request from '../request'

export interface Doctor {
  doctorID: number
  doctorName: string
  doctorGender: number
  doctorPhone: string
  doctorEmail?: string
  departmentID: number
  departmentName?: string
  title: string
  specialty?: string
  doctorIntro?: string
  doctorPhoto?: string
  registrationFee: number
  doctorStatus: number
}

export const doctorAPI = {
  /**
   * 获取医生列表
   */
  getList: () => {
    return request.get<Doctor[]>('/hospital/doctor/list')
  },

  /**
   * 获取医生详情
   */
  getById: (id: number) => {
    return request.get<Doctor>(`/hospital/doctor/${id}`)
  },

  /**
   * 根据科室获取医生列表
   */
  getByDepartment: (departmentId: number) => {
    return request.get<Doctor[]>(`/hospital/doctor/department/${departmentId}`)
  },

  /**
   * 根据职称获取医生列表
   */
  getByTitle: (title: string) => {
    return request.get<Doctor[]>(`/hospital/doctor/title/${title}`)
  },

  /**
   * 搜索医生
   */
  search: (keyword: string) => {
    return request.get<Doctor[]>('/hospital/doctor/search', { keyword })
  },

  /**
   * 医生登录
   */
  login: (phone: string, password: string) => {
    return request.post<Doctor>('/hospital/doctor/login', null, { params: { phone, password } })
  },

  /**
   * 添加医生
   */
  add: (data: Partial<Doctor>) => {
    return request.post<string>('/hospital/doctor', data)
  },

  /**
   * 更新医生信息
   */
  update: (id: number, data: Partial<Doctor>) => {
    return request.put<string>(`/hospital/doctor/${id}`, data)
  },

  /**
   * 删除医生
   */
  delete: (id: number) => {
    return request.delete<string>(`/hospital/doctor/${id}`)
  },

  /**
   * 更新医生状态
   */
  updateStatus: (id: number, status: number) => {
    return request.put<string>(`/hospital/doctor/${id}/status`, { status })
  }
}
